crates/metrics/src/txpool_metrics.rs

```diff
@@ -1,24 +1,97 @@
-use crate::global_registry;
-use prometheus_client::metrics::histogram::Histogram;
+use crate::{
+    buckets::{
+        buckets,
+        Buckets,
+    },
+    global_registry,
+};
+use prometheus_client::metrics::{
+    gauge::Gauge,
+    histogram::Histogram,
+};
 use std::sync::OnceLock;
 
 pub struct TxPoolMetrics {
-    pub tx_size_histogram: Histogram,
+    /// Size of transactions in the txpool in bytes
+    pub tx_size: Histogram,
+    /// Number of transactions in the txpool
+    pub number_of_transactions: Gauge,
+    /// Number of transactions pending verification, before being inserted in the txpool
+    pub number_of_transactions_pending_verification: Gauge,
+    /// Number of transactions that can be included in the next block
+    pub number_of_executable_transactions: Gauge,
+    /// Time of transactions in the txpool in seconds
+    pub transaction_time_in_txpool_secs: Histogram,
+    /// Time actively spent by transaction insertion in the thread pool
+    pub transaction_insertion_time_in_thread_pool_milliseconds: Histogram,
+    /// How long it took for the selection algorithm to select transactions
+    pub select_transaction_time_nanoseconds: Histogram,
```

Comment:
I think it is better to use the same units to track things. Like either nanoseconds, or milliseconds, or seconds. Not all 3=D

I think Nanoseconds are ok, we always can divide it=)

Reply:
I'm ok with having different units in the metric producer (I mean: units that fit the value being observed best), as long as the unit is clearly communicated in the metric name. The presentation layer could do all necessary shenanigans required to show proper data.

Otoh, there's also a merit in having a consistent time unit, but nanoseconds everywhere are too fine-grained in my opinion. For example, if transactions can live up to 10 minutes in the TxPool we're gonna be dealing with numbers such as `600_000_000_000`.

I'll wait for more opinions before I start fiddling with the code.

Reply:
I don't have a strong opinion on whether we decide to stick with nanoseconds for all metrics, as @rafal-ch said it has the merit of bringing uniformity across metrics.  But even in that case we will need multiple timing buckets for different granularities.

Reply:
As per the meeting: 
`transaction_insertion_time_in_thread_pool` and `select_transaction_time` should be measured in microseconds.
10us - minimal value, 5s - max value (however, unlikely)

we don't aim for a single unit for all metrics at the moment.

Reply:
Changes in https://github.com/FuelLabs/fuel-core/pull/2385/commits/49a79b2ec0fca0da5e5ab1317afbd0ab230b8a93